### PR TITLE
Warn about port forwarding

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

File: docs/getting-started/local_instance.md [hunk 1]
```diff
@@ -29,9 +29,12 @@ python butler.py run_server --bootstrap
 python butler.py run_server
 ```
 
-This may take a few seconds to start. Once you see output line like
-`INFO <timestamp> admin_server.py:<num>] Starting admin server`, you should be able to navigate to
-[http://localhost:9000](http://localhost:9000) to view the web interface.
+This may take a few seconds to start. Once you see an output line like
+`INFO <timestamp> admin_server.py:<num>] Starting admin server`, you should be
+able to navigate to [http://localhost:9000](http://localhost:9000) to view the
+web interface. Note that the local instance may use ports other than 9000, such
```
Comment (inferno-chromium) [inline docs/getting-started/local_instance.md]: nit: can you use Note to next line as

<star><star>Note<star><star>:

Comment (inferno-chromium) [inline docs/getting-started/local_instance.md]: Can you hyperlink "other than 9000" to https://github.com/google/clusterfuzz/blob/master/src/local/butler/constants.py ? that has list of ports.

File: docs/getting-started/local_instance.md [hunk 2]
```diff
@@ -29,9 +29,12 @@ python butler.py run_server --bootstrap
...
+as 9008, for things like uploading files. Therefore, using port forwarding to
+access a local instance may break.
```
Comment (inferno-chromium) [inline docs/getting-started/local_instance.md]: "break if all the needed ports are not forwarded" or something like that ?

**Comment by jonathanmetzman** [comment]
Should fix #163
cc @kcwu @charleyw

File: docs/getting-started/local_instance.md (hunk 1 above)
Comment (jonathanmetzman) [reply docs/getting-started/local_instance.md]: Done.

File: docs/getting-started/local_instance.md (hunk 2 above)
Comment (jonathanmetzman) [reply docs/getting-started/local_instance.md]: Done.

File: docs/getting-started/local_instance.md (hunk 1 above)
Comment (jonathanmetzman) [reply docs/getting-started/local_instance.md]: Done.

**Comment by jonathanmetzman** [comment]
@inferno-chromium I made this about more than port forwarding since that is a pretty specific case.
I think a better fix is to fail loudly on the client side if the upload fails.

File: docs/getting-started/local_instance.md [hunk 3]
```diff
@@ -32,9 +32,11 @@ python butler.py run_server
 This may take a few seconds to start. Once you see an output line like
 `INFO <timestamp> admin_server.py:<num>] Starting admin server`, you should be
 able to navigate to [http://localhost:9000](http://localhost:9000) to view the
-web interface. Note that the local instance may use ports other than 9000, such
-as 9008, for things like uploading files. Therefore, using port forwarding to
-access a local instance may break.
+web interface.
+Note: the local instance may use ports [other than 9000](https://github.com/google/clusterfuzz/blob/master/src/local/butler/constants.py),
+such as 9008, for things like uploading files. Therefore, using the local
```
Comment (kcwu) [inline docs/getting-started/local_instance.md]: How about generalize this, ask users to tweak network rules like firewall and port forwarding.

Comment (jonathanmetzman) [reply docs/getting-started/local_instance.md]: I think mentioning port forwarding and allowing through firewall is kind of too specific and I don't want to complicate the docs for the majority of users for whom this isn't an issue. What do you think of changing the last sentence to this:
>Therefore, using the local instance may break if the needed ports are unavailable or if you can only access some of the needed ports from your browser.

Comment (inferno-chromium) [reply docs/getting-started/local_instance.md]: sgtm

Comment (jonathanmetzman) [reply docs/getting-started/local_instance.md]: I ended up warning about port forwarding and firewall

**Pull request merged**